main.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    (0..length - 1).each do |i|
+      yield(self[i])
+    end
+  end
+
+  def my_map(&proc)
+    temp = []
+    (0..length - 1).each do |i|
+      temp << [proc.call(self[i])] if proc
+      temp << yield(self[i]) if block_given? && !proc
+    end
+    temp
+  end
+
+  def my_each_with_index
+    (0..length - 1).each do |i|
+      yield(self[i], i)
+    end
+  end
+
+  def my_select
+    temp = []
+    (0..length - 1).each do |i|
+      temp << self[i] if yield(self[i])
+    end
+    temp
+  end
+
+  def my_any?
+    each do |item|
+      return true if block_given? && yield(item)
+      return true if item && !block_given?
+    end
+    false
+  end
+
+  def my_all?
+    each do |item|
+      if block_given?
+        return false unless yield(item)
+      else return false unless item
+      end
+    end
+    true
+  end
+
+  def my_none?(condition = false)
+    each do |item|
+      if block_given?
+        return false if yield(item)
+      elsif item == condition then return false
+      end
+    end
+    true
+  end
+
+  def my_count(cond = true)
+    counter = 0
+    each do |item|
+      if block_given? && yield(item)
+        counter += 1
+      elsif item == cond
+        counter += 1
+      end
+    end
+    counter
+  end
+
+  def my_inject(acc = 0)
+    each do |item|
+      acc = yield(acc, item)
+    end
```

Comment:
- [ ] my_inject combines all elements of enum by applying a binary operation, specified by a block
- [ ] my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method